### PR TITLE
tests: Define tests/unit folder

### PR DESCRIPTION
**Pull request by frascuchon**
# Description
⚠️ This PR must be merged after #3482

This PR defines the `tests/unit` folder, which includes unit tests for the server and client part. Those tests will check single component responsibility, as and specific endpoint functionality, or model schema validation. and won't contain tests checking more complex workflows or checking client-server integrations.

The old tests locations will be removed in future PRs

### CODE REVIEW

File: tests/unit/client/feedback/training/test_trainer.py
Comment (tomaarsen) [inline tests/unit/client/feedback/training/test_trainer.py:1]: Is there a motivation for including this file in `unit` while the feedback trainer tests are in `integration`?

Comment (tomaarsen) [reply tests/unit/client/feedback/training/test_trainer.py:1]: For the rest I think the split is pretty sensible

Comment (frascuchon) [reply tests/unit/client/feedback/training/test_trainer.py:1]: Wrong place. This should be part of the integration tests. Thanks!

**Review by tomaarsen — APPROVED** [review]
This split seems sensible to me. I don't know if there's any documentation or something that points to any test files, if so, we'd have to update them.

**Comment by frascuchon** [comment]
> This split seems sensible to me. I don't know if there's any documentation or something that points to any test files, if so, we'd have to update them.

I will review docs later in a separate PR

**Pull request merged**